### PR TITLE
 osd: add hdd, ssd and hybrid variants for osd_snap_trim_sleep

### PR DESCRIPTION
**Pull request by neha-ojha**
Similar to `osd_delete_sleep`, having better defaults for `osd_snap_trim_sleep` will be useful.

Fixes: https://tracker.ceph.com/issues/40528
Signed-off-by: Neha Ojha <nojha@redhat.com>

- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: doc/rados/configuration/osd-config-ref.rst [hunk 1]
```diff
@@ -494,6 +494,40 @@ Operations
 :Default: ``5``
 :Valid Range: 1-63
 
+``osd snap trim sleep``
+
+:Description: Time in seconds to sleep before next snap trim op.
+              Increasing this value will slow down snap trimming.
```
Comment (dzafman) [inline doc/rados/configuration/osd-config-ref.rst]: Should this mention that setting this value supersedes the settings below?

Comment (neha-ojha) [reply doc/rados/configuration/osd-config-ref.rst]: I believe that is true for all the sleep values, added a commit to be more specific

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-06-28_13:46:32-rados-wip-kefu-testing-2019-06-28-1816-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-07-02_06:19:52-rados-wip-kefu-testing-2019-06-28-1816-distro-basic-mira/

the failures in upgrade tests are caused by #28766 and i don't think the tasks.mgr.test_progress.TestProgress failure is related.

**Comment by tchaikov** [comment]
```
[ RUN      ] chain_xattr.get_and_set

[WARNING] /home/jenkins-build/build/workspace/ceph-pull-requests/src/googletest/googletest/src/gtest-death-test.cc:1112:: Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test detected 3 threads. See https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#death-tests-and-threads for more explanation and suggested solutions, especially if this is the last message you see before your test times out.


99% tests passed, 1 tests failed out of 178

Total Test time (real) = 3895.71 sec

The following tests FAILED:
	133 - unittest_chain_xattr (Timeout)
```

this test has been failing recently...

retest this please.

**Comment by neha-ojha** [comment]
retest this please

**Pull request merged**